mp/src/game/server/filters.cpp

```diff
@@ -324,6 +326,40 @@ BEGIN_DATADESC( FilterTeam )
 END_DATADESC()
 
 
+// ###################################################################
+//	> FilterNeoClass
+// ###################################################################
+class FilterNeoClass : public CBaseFilter
+{
+	DECLARE_CLASS(FilterNeoClass, CBaseFilter);
+	DECLARE_DATADESC();
+
+public:
+	int m_iFilterNeoClass;
+
+	bool PassesFilterImpl(CBaseEntity* pCaller, CBaseEntity* pEntity)
+	{
+		if (!pEntity || !pEntity->IsPlayer())
+			return false;
+
+		CNEO_Player* pPlayer = dynamic_cast<CNEO_Player*>(pEntity);
+		if (!pPlayer)
+			return false;
```

Comment:
Using `static_cast` (or `assert_cast`), this pointer *must* be valid. So we can skip this check entirely.